lib/widgets/message_list.dart

```diff
@@ -924,8 +931,12 @@ class DateSeparator extends StatelessWidget {
 
     final line = BorderSide(width: 0, color: messageListTheme.dateSeparator);
 
-    // TODO(#681) use different color for DM messages
-    return ColoredBox(color: messageListTheme.streamMessageBgDefault,
+    final backgroundColor = switch (message) {
+      StreamMessage() => messageListTheme.streamMessageBgDefault,
+      DmMessage() => messageListTheme.dmMessageBgDefault,
+    };
```

Comment:
Let's actually define a helper function with this logic, to use in both `DateSeparator` and `MessageItem`. The background-color logic will get more complicated with #647, and we'll want the date-separator and message-item logic to stay in sync when that happens.